packages/safe-core-sdk/src/Safe.ts

```diff
@@ -260,11 +261,17 @@ class Safe {
   ): Promise<SafeTransaction>
   async createTransaction(
     safeTransactions: SafeTransactionDataPartial | MetaTransactionData[],
-    options?: SafeTransactionDataPartial
+    options?: SafeTransactionOptionalProps
   ): Promise<SafeTransaction> {
-    if (safeTransactions instanceof Array) {
+    const isTxArray = Array.isArray(safeTransactions)
+    if (isTxArray && (safeTransactions as MetaTransactionData[]).length === 0) {
```

Comment:
Will this always be `MetaTransactionsData[]`?

Reply:
As well as the other casts you make in this function?

Reply:
You could also just cast it once under a new variable, no?

Reply:
It can be `MetaTransactionsData[]` or `SafeTransactionDataPartial`

Reply:
Maybe make a type checking function that returns `value is X`?